pkg/controllers/registryclient/controller_test.go

```diff
@@ -79,44 +80,59 @@ var _ = Describe("ClientController", func() {
 
 		remoteRegistry = &connectivityv1alpha1.RemoteRegistry{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      "some-remote-registry",
-				Namespace: "cross-cluster-connectivity",
+				Name:       "some-remote-registry",
+				Namespace:  "cross-cluster-connectivity",
+				UID:        "remote-registry-uid",
+				Generation: 1,
 			},
 			Spec: connectivityv1alpha1.RemoteRegistrySpec{
 				Address: fmt.Sprintf("127.0.0.1:%d", hamletPort),
 			},
 		}
 
-		_, err = connClientset.ConnectivityV1alpha1().RemoteRegistries("cross-cluster-connectivity").
+	})
+
+	JustBeforeEach(func() {
+		_, err := connClientset.ConnectivityV1alpha1().RemoteRegistries("cross-cluster-connectivity").
 			Create(remoteRegistry)
 		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() (*connectivityv1alpha1.RemoteRegistry, error) {
+			return connClientset.ConnectivityV1alpha1().RemoteRegistries("cross-cluster-connectivity").
+				Get(remoteRegistry.Name, metav1.GetOptions{})
+		}, 5*time.Second, time.Second).Should(MatchObservedGeneration(remoteRegistry.Generation))
```

Comment:
We depend on the updating of the Observed Generation to prevent hot looping. We should have a test (an explicit "It" that will fail and tell me as a dev why was this important) that the Observed Generation gets updated. 

Tyler and I were discussing if adding a test to check that the generation bump happens only once per update is a possible alternative. We think not because
1. It feels weird to test the absence of logic. 
2. We are not running a real kube api, but a fake. Any reproduction of the hot loop would be emulated, and not a real test.

Reply:
That's a good point, I'll try to make this intent clearer.